lib/ansible/modules/files/find.py

```diff
@@ -255,14 +255,12 @@ def contentfilter(fsname, pattern):
         return True
 
     try:
-        f = open(fsname)
-        prog = re.compile(pattern)
-        for line in f:
-            if prog.match(line):
-                f.close()
-                return True
-
-        f.close()
+        with open(fsname) as f:
+            prog = re.compile(pattern)
```

Comment:
We should move the `re.compile` out of the context manager.  Potentially even before the `open`.

Also we should catch invalid regex patterns separate from issues opening the file.